libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/FormatOptionsDropdown.tsx

```diff
@@ -0,0 +1,149 @@
+
+    const canApply =
+        format !== props.format ||
+        mergeHeaders !== props.mergeHeaders ||
+        includeFilters !== props.includeFilters;
+
+    const handleOnApply = useCallback(() =>
+        onApply({
+            format,
+            mergeHeaders,
+            includeFilters,
+        }), [format,mergeHeaders,includeFilters]
+    );
+
+    const handleOnCancel = useCallback(() => {
+        setFormat(props.format);
+        setMergeHeaders(props.mergeHeaders);
+        setIncludeFilters(props.includeFilters);
+    }, [props.format, props.mergeHeaders, props.includeFilters]);
+
+    const renderBodyContentSelector = () => {
+        const insightFormat = intl.formatMessage({ id: "dialogs.schedule.email.insight.format" }) + ":";
+        return (
+            <div className="gd-format-options-dropdown-selector">
+                <span className="input-label-text">{insightFormat}</span>
+                <div>
+                    <Dropdown
+                        overlayPositionType="sameAsTarget"
+                        className="s-format-options-dropdown-ignore"
+                        alignPoints={DEFAULT_DROPDOWN_ALIGN_POINTS}
+                        overlayZIndex={DEFAULT_DROPDOWN_ZINDEX}
+                        renderButton={({ isOpen, toggleDropdown }) => (
+                            <DropdownButton
+                                value={format.toUpperCase()}
+                                isOpen={isOpen}
+                                onClick={toggleDropdown}
+                            />
+                        )}
+                        renderBody={({ closeDropdown }) => (
+                            <DropdownList
+                                width={DROPDOWN_WIDTH}
+                                items={FORMAT_VALUES}
+                                renderItem={({ item }) => {
+                                    return (
+                                        <SingleSelectListItem
+                                            className="s-format-options-dropdown-content-ignore"
+                                            title={item.toUpperCase()}
+                                            onClick={() => {
+                                                setFormat(item);
+                                                closeDropdown();
+                                            }}
+                                        />
+                                    );
+                                }}
+                            />
+                        )}
+                    />
+                </div>
+            </div>
+        );
+    };
+
+    const renderBodyContentSelectorOptions = () => {
+        const attributeMergedMessage = intl.formatMessage({ id: "dialogs.schedule.email.attribute.merged" });
+        const showFiltersMessage = intl.formatMessage({ id: "dialogs.schedule.email.show.filters" });
```

Comment:
These could be replaced by `FormattedMessage` for simplicity. Also one more above